### ISSUE TITLE
Provide a way to isolate tests within one document

### ISSUE THREAD

**Issue by adamtheturtle** [simplistix/sybil#47]
I am trying to switch from `doctest` to `code-block` with Sybil.
One issue I'm hitting is that I would like to have multiple separated tests within one document.

[`manuel` provides](https://pythonhosted.org/manuel) `.. reset-globs`.
Is there a workaround to allow similar functionality with Sybil?

**Comment by adamtheturtle** [comment]
I'm using:

```restructuredtext
.. invisible-code-block: python

   # Reset test
   for i in list(globals().keys()):
    if not i.startswith('_'):
        exec('del ' + i)
```

but this seems quite hacky.
My ideal is probably something like the `group`s in https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html.

**Comment by cjw296** [comment]
Feels like Sybil should grow something :-)

Rather than hacking the above, why not add your own parser that does this?
https://sybil.readthedocs.io/en/latest/use.html#developing-your-own-parsers

**Comment by adamtheturtle** [comment]
(and, thank you for the very neat project!)

**Comment by cjw296** [comment]
@adamtheturtle - what Python version are you using?

**Comment by adamtheturtle** [comment]
@cjw296 Python 3.11.2

**Comment by adamtheturtle** [comment]
@cjw296 pytest 7.2.2

**Comment by cjw296** [comment]
Implemented in 8012155168fa4a088502ab59e309a5e9f56d16fe

**Comment by cjw296** [comment]
Released in https://pypi.org/project/sybil/5.0.0/
